skbeam/core/roi.py

```diff
@@ -458,6 +459,67 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     return bin_centers, ring_averages
 
 
+def construct_circ_avg_image(radii, intensities, center=None, dims=None,
+                             pixel_size=(1, 1)):
+    """ Constructs a 2D image from circular averaged data
+        where radii are given in units of pixels.
+        Normally, data will be taken from circular_average and used to
+        re-interpolate into an image.
+
+    Parameters
+    ----------
+    radii : 1D array of floats
+        the radii (must be in pixels)
+    intensities : 1D array of floats
+        the intensities for the radii
+    center: 2 tuple of floats, optional
+        [y0, x0] (row, col)
+        y0, x0 is the center (in row,col format)
+        dims also needs be set if using this option
+    dims : 2 tuple of floats, optional
+        [dy, dx] (row, col)
+        dy, dx are the dimensions in row,col format
+        If either center or dims are not set, it will assume the dimensions to
+        be twice the maximum radius and the center to be the center of the
+        image: (img.shape[0]-1)/2., (img.shape[1]-1)/2.)
+
+    pixel_size : tuple, optional
+        The size of a pixel (in a real unit, like mm).
+        argument order should be (pixel_height, pixel_width)
+        default is (1, 1)
+
+    Returns
+    -------
+    IMG : the interpolated circular averaged image
+
+    See Also
+    --------
+    circular_average : compute circular average of an image
+    bin_grid : Bin and integrate an image, given the radial array of pixels
+        Useful for nonlinear spacing (Ewald curvature)
+
+    Notes
+    -----
+    Some pixels may not be filled if the dimensions chosen are too large.
+        Run this code again on a list of values equal to 1 to obtain a mask.
+
+    Example
+    -------
+    """
+    if center is None or dims is None:
```

Comment:
that makes sense. `center` but no `dims` does not however. I'll raise an exception there just in case.